CBFlib 0.9.7 devel

yayahjb opened the pull request:
This is the 0.9.7 development branch being proposed for a new 0.9.7 release.  One recent major change is the restructuring of pycbf.  The old pycbf using python2 is now py2cbf and pycbf has been restructured to use python 3.  There is one unexplained warning in pycbf_test1.

yayahjb committed:
Fix pycbf tests to run without error and check all output
Use numpy instead of Number --HJB

yayahjb committed:
Merge branch 'main' into CBFlib-0.9.7-devel

yayahjb committed:
Update Makefiles -- HJB

yayahjb committed:
Clean up pycbftests and pycbftests_sigs_only -- HJB

yayahjb committed:
Change img version of memcpy to resolve gcc-10 issues in catalina
Update more Makefiles with boz flag needed for gfortran-10

yayahjb committed:
Clean up Makefiles and make distclean except for old Makefiles -- HJB

yayahjb committed:
Only make Makefile, Makefile_LINUX Makefile_MINGW Makefile_MSYS2 Makefile_OSX m4/Makefile.m4

yayahjb committed:
Fix missing -f on some make empty items

yayahjb committed:
Use int casts to suppress BOZ warnings for gfortran -- HJB

yayahjb committed:
Reverse sense of test on line 1206 of testhdf5.c so default is all tests pass -- HJB

yayahjb committed:
Merge branch 'main' into CBFlib-0.9.7-devel

yayahjb committed:
Clean up misc gcc warnings -- HJB

yayahjb committed:
Clean up third dimension errors -- HJB

yayahjb committed:
Fix gcc-11 error for lz4 plugin
Fix coding error in cbf_H5Dcreate revealed by hdf 1-12
  -- HJB

yayahjb committed:
Update version number to 0.9.7 in cbf.h
Fix tests to reflect correct version number
Suppress fatal warning about missing static function type

yayahjb committed:
Remove some unused variables

yayahjb committed:
Fix test errors by dropping c1 and c2 compressions

yayahjb committed:
Change pycbf to py3cbf

yayahjb committed:
move pycbf to py2cbf

yayahjb committed:
Bring old pycbf into repository as py3cbf with history
Merge remote-tracking branch 'repo2/main' into CBFlib-0.9.7-devel

yayahjb committed:
Incomplete py3cbf -- HJB

yayahjb committed:
Turn off py3cbf until outstanding issues are resolved -- HJB

yayahjb committed:
Patch encoding issue with python 2.7.13 and pycbf.py

yayahjb committed:
Clean up py2cbf to explicitly rebuild from nuweb and provide a
nuweb kit.  Similar update to py3cbf will follow. -- HJB

yayahjb committed:
As agreed move py3cbf back to pycbf
Also change py2cbf _pycbf module to _py2cbf

yayahjb committed:
Partial cleanup for python 3 pycbf tests.  Had to remove -py3.  More to come.

yayahjb committed:
Clean up py3cbftests to that there is only one warning

yayahjb committed:
Update to cif_img_1.8.5 -- HJB

yayahjb committed:
Correct some periods that should be underscores in 1.8.5 dictionary
draft

Commit:
Changes to fix tests in dials.lbl.gov environment, added
Makefile_DIALS which still needs to be merged into m4/Makefile.m4
added a local copy of tif_sprint.c to fix a linking problem

yayahjb commented:
OK, I have updated the yayahjb cbflib 0.9.7 branch and then synced the cbflib cbflib branch
and tested the updated version using this initialization:

source /dev/shm/aaron/build/conda_setpaths.sh # this loads a python3 build of cctbx with hdf5lib.
LD_LIBRARY_PATH=/dev/shm/aaron/conda_base/lib
export LD_LIBRARY_PATH
HDF5_PREFIX=/dev/shm/aaron/conda_base
export HDF5_PREFIX

and using

git clone http://github.com/cbflib/cbflib.git
git checkout CBFlib-0.9.7-devel
make -f Makefile_DIALS distclean
make -f Makefile_DIALS tests

I still have to merge the changes in Makefile_DIALS into m4/Makefile.m4
but I think I should wait on that until I see what issues you two have.
I also need to add more documentation to the release.

Regards,
    Herbert

P.S.  Your /dev/shm files seem to work nicely

phyy-nx commented:
Hi, I followed the above steps and the 0.9.7 branch passed all its tests.  That's excellent.

I also ran the dials and dxtbx tests using this branch and they all passed.

What I'd like to do now is to try converting some of the cbf reading code in dxtbx that uses byte  strings to unicode strings, and the code that reads data to use the new _asbytes functions.  I'll report back when I have results.

yayahjb commented:
Dear Aaron,
 You made my morning.
  Regards,
    Herbert

On Mon, Jul 12, 2021 at 8:40 PM Aaron S. Brewster ***@***.***>
wrote:

> Hi, I followed the above steps and the 0.9.7 branch passed all its tests.
> That's excellent.
>
> I also ran the dials and dxtbx tests using this branch and they all passed.
>
> What I'd like to do now is to try converting some of the cbf reading code
> in dxtbx that uses bytest strings to unicode strings, and the code that
> reads data to use the new _asbytes functions. I'll report back when I have
> results.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cbflib/cbflib/pull/1#issuecomment-878690050>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABB6EAJIYGZVEDNY4Q7ITZDTXODORANCNFSM47XVVBLQ>
> .
>

yayahjb committed:
Update doc to cif_img_1.8.5

yayahjb committed:
Merge pull request #41 from yayahjb/CBFlib-0.9.7-devel

Cbflib 0.9.7 devel

PeterC-DLS committed:
Fix warnings about long integers for win32 64-bit builds

PeterC-DLS committed:
Include cbf_drel in header file only if also compiled in library

yayahjb committed:
Merge pull request #43 from DiamondLightSource/fix-long-warnings

Fix warnings about long integers for win32 64-bit builds

PeterC-DLS committed:
Update Java bindings build

Load main library before wrapper in test, use JDK's java launcher, fix macOS build to include platform specific JNI header and update to compile with C99 standard

yayahjb committed:
Merge pull request #45 from DiamondLightSource/update-java-build

Update Java bindings build

yayahjb committed:
Merge pull request #42 from DiamondLightSource/fix-extra-definition

Include cbf_drel in header file only if also compiled in library

yayahjb committed:
Proposed revised dictionary 1.8.6 for JRH pull request -- HJB

PeterC-DLS committed:
Make shared library extension configurable

PeterC-DLS committed:
Add MingW cross-compile

yayahjb committed:
Fix misplaced period in example tag as per JRH

yayahjb committed:
Add cif_img_1.8.6 dictionary from JRH

yayahjb committed:
Merge pull request #48 from DiamondLightSource/shared-lib-extension

Make shared library extension configurable

yayahjb committed:
Merge pull request #49 from DiamondLightSource/mingw-crosscompile

Add Makefile for cross-compiling with MinGW

yayahjb committed:
Update cif_img_1.8.6.dic

yayahjb committed:
Update cif_img_1.8.6.dic

yayahjb committed:
As per JRH fix trailing blanks in quoted tag and make
_array_data.external_data_id implicit

yayahjb committed:
Add latest JRH fixes
Merge branch 'main' into CBFlib-0.9.7-devel

yayahjb committed:
Update cif_img_1.8.6.dic

Fix missing loop_ as per JRH

yayahjb committed:
Update cif_img_1.8.6.dic

Fix missing loop_ as per JRH

yayahjb committed:
Fix various typos in cif_img_1.8.6.dic and start conforming
cif_img_1.8.6.html to cif_img_1.8.6.dic.

yayahjb committed:
Update to nuweb 1.61 to avoid ld errors on Debian 11 -- HJB

yayahjb committed:
Fix py2cbf and pycbf install logic in Makefile

yayahjb committed:
Merge branch 'main' into CBFlib-0.9.7-devel

yayahjb committed:
Update LINUX and default hdf5 builds to use fPIC and -lpthread -- HJB

yayahjb committed:
Add verbose and help options and fix elsize calculation
